feat(automated testing): Flow Container tests added

pomi opened the pull request:
## Changes
Flow Container tests are added

Commit:
Migration from cypress to playwright, updated

Commit:
Flow Container test is added

Commit:
Obsolete comments are deleted

Commit:
Merge branch 'pw_poc' into AESQ-2883

Commit:
Merge branch 'main' into AESQ-2883

Commit:
preliminary commit

Commit:
Files updates to comply with MR comments

Commit:
Files updates to comply with MR comments

Commit:
preliminary commit

Commit:
base page was updated to resolve conflicts

TatsianaMiraniuk commented on sites/test-site/src/data/pages/flow-container/index.tsx:
```diff
@@ -185,11 +185,19 @@ const FlowContainerPage = (props: any) => (
       />
       <H2>Contentful Flow Container</H2>
-      <ContentfulFlowContainer />
+      <ContentfulFlowContainer
```
@pomi ,

the IDs are added, but they are not used.

TatsianaMiraniuk commented on playwright/pages/flow-container-page.ts:
```diff
@@ -0,0 +1,83 @@
+    this.flowContainer50Width = '#width_50';
+    this.flowContainer66Width = '#width_66';
+    this.flowContainer75Width = '#width_75';
+    this.flowContainerContentful = '#gatsby-focus-wrapper > div:nth-child(1) > div.my-2.container.mx-auto > div:nth-child(32) > section > div.bl-resizable.bl-relative.bl-border-transparent.bl-border-2.bl-border-dashed.hover\\:bl-border-primary.w-full.md\\:p-5.bl-relative';
```
@pomi ,

the locators should be updated to more stable. 
3 IDs have been added in sites/test-site/src/data/pages/flow-container/index.tsx, but they are not used. Please use them, and request others if needed to make the locators stable and resilient to the changes.

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    await page.click(flowContainerPage.squareImageButtonInsideContainer);
+    await page.click(flowContainerPage.selectImageButton);
+    await page.setInputFiles('input[type=file]', flowContainerPage.pathToImages + flowContainerPage.imageTwoName);
+    await page.fill('#image-alt', 'alt-test');
```
@pomi ,

please, move to the variables instead of hardcoding

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    ]);
+    const defaultContainer = page.locator(flowContainerPage.flowContainerDefault);
+    const img = defaultContainer.locator('img');
+    expect.soft(await img.getAttribute('alt')).toEqual('alt-test');
```
the variable should be used instead of hardcoding

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+  let page: Page;
+  let flowContainerPage: FlowContainerPage;
+  test.beforeAll(async ({ browser }) => {
+    const context = await browser.newContext({ viewport: { width: 1200, height: 850 } });
```
@pomi ,

can be deleted from a test level as it's available on a global level

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    const context = await browser.newContext({ viewport: { width: 1200, height: 850 } });
+    page = await context.newPage();
+    flowContainerPage = new FlowContainerPage(page);
+    await page.goto('/flow-container');
```
@pomi,

a trailing slash in URL is expected.

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+        .includes('width_33') && response.status() === 200),
+      await page.click(flowContainerPage.accordionInPicker),
+    ]);
+    const containerWidth = await page.locator('#gatsby-focus-wrapper > div:nth-child(1) > div.my-2.container.mx-auto > div:nth-child(24) > section').boundingBox();
```
@pomi ,

please move a locator value to a variable

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    expect.soft(await page.locator(flowContainerPage.deleteComponentButton).isVisible()).toBeFalsy();
+    const defaultContainer = page.locator(flowContainerPage.flowContainerDefault);
+    const img = defaultContainer.locator('img');
+    expect.soft(await img.getAttribute('alt')).toEqual('alt-test');
```
@pomi ,

a variable should be used instead of hardcoding

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+
+  test('checking swapping a component (Square Image to Landscape Linkable Image)', async () => {
+    await flowContainerPage.toggleEditMode();
+    await page.locator('#flowContainer > div > img').click();
```
@pomi ,

a variable should be used instead of hardcoding

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    await page.locator('#flowContainer > div > img').click();
+    await page.click(flowContainerPage.swapComponentButton);
+    await page.click(flowContainerPage.contentfulCheckbox);
+    await page.click('div[data-item-id="LandscapeLinkableImage"]');
```
@pomi ,

a variable should be used instead of hardcoding

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+  test('checking the swapped and deleted components in Preview Mode', async () => {
+    await flowContainerPage.togglePreviewMode();
+    await page.locator('#flowContainer > div a >> img').click();
+    expect.soft(page.url()).toContain('/' + 'new_link' + '/');
```
@pomi ,

please replace toContain with toEqual

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    expect.soft(ratio).toBeCloseTo(32);
+  });
+
+  test('checking the added components in Preview Mode', async () => {
```
@pomi 

numeration is missed in the name

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    expect.soft(await page.locator('#width_33 > div').isVisible()).toBeTruthy();
+  });
+
+  test('checking swapping a component (Square Image to Landscape Linkable Image)', async () => {
```
@pomi 

numeration is missed in the name

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    expect.soft(await page.locator('#flowContainer > div a').getAttribute('href')).toEqual('/' + 'new_link' + '/');
+  });
+
+  test('checking the swapped and deleted components in Preview Mode', async () => {
```
@pomi 

numeration is missed in the name

TatsianaMiraniuk replied:
@pomi ,

the test for deleting a component is missing.
First a component should be deleted in Edit mode, in a new test prior tho this one.
And then it has to be confirmed that the component is not available on Preview mode (in the current test)

TatsianaMiraniuk commented on playwright/pages/flow-container-page.ts:
```diff
@@ -0,0 +1,83 @@
+    this.deleteComponentButton = 'button[aria-label="Delete Component"]';
+    this.selectImageButton = 'button[aria-label="Select Image"]';
+    this.editImageButton = 'button[aria-label="Edit Image"]';
+    this.squareImageButton = '#gridlistboxinner > div:nth-child(1) > div > button';
```
the locator should be updated to more stable.

TatsianaMiraniuk commented on playwright/pages/flow-container-page.ts:
```diff
@@ -0,0 +1,83 @@
+    this.accordionPlusButton = 'span[aria-label="Expand Accordion"]';
+    this.accordionMinusButton = 'span[aria-label="Collapse Accordion"]';
+    this.contentfulCheckbox = '#Contentful';
+    this.accordionBody = 'div[data-bl-design-key="Accordion:Body"] div div div';
```
the locator should be updated to more stable.

TatsianaMiraniuk commented on playwright/pages/flow-container-page.ts:
```diff
@@ -0,0 +1,83 @@
+    this.contentfulCheckbox = '#Contentful';
+    this.accordionBody = 'div[data-bl-design-key="Accordion:Body"] div div div';
+    this.accordionInsideRestrictedContainer = 'div[id^="accordion-"]';
+    this.squareImageButtonInsideContainer = '#gatsby-focus-wrapper > div:nth-child(1) > div.my-2.container.mx-auto > div:nth-child(4) > section > div > img';
```
the locator should be updated to a stable one.

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    expect.soft(await page.locator(flowContainerPage.swapComponentButton).isVisible()).toBeTruthy();
+    expect.soft(await page.locator(flowContainerPage.deleteComponentButton).isVisible()).toBeTruthy();
+    expect.soft(await page.locator(flowContainerPage.addComponentButton).isVisible()).toBeTruthy();
+  });
```
@pomi ,

a check for Copy button is missing

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    expect.soft(await page.locator(flowContainerPage.accordionBody).isVisible()).toBeFalsy();
+    await page.click(flowContainerPage.accordionPlusButton);
+    expect.soft(await page.locator(flowContainerPage.accordionBody).isVisible()).toBeTruthy();
+    expect.soft(await page.locator('#width_33 > div').isVisible()).toBeTruthy();
```
@pomi,

#width_33 > div is valid locator for an empty FlowContainer too - so it doesn't prove that Contentful Card is visible on Preview.
Please add a check that confirm Contentful Card is visible

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    await page.click(flowContainerPage.accordionInPicker);
+    await page.click(flowContainerPage.accordionInsideRestrictedContainer);
+    await flowContainerPage
+      .typeText(flowContainerPage.accordionInsideRestrictedContainer, 'text to check', 'restricted');
```
should be moved to a variable

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    await flowContainerPage
+      .typeText(flowContainerPage.accordionInsideRestrictedContainer, 'text to check', 'restricted');
+    expect.soft(await page.locator(flowContainerPage.accordionInsideRestrictedContainer)
+      .innerText()).toEqual('text to check');
```
should be moved to a variable

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    // text is not typed without this timeout, race condition
+    await page.waitForTimeout(300);
+    await flowContainerPage
+      .typeText(flowContainerPage.accordionBody, 'accordion-body-text', '$body');
```
should be moved to a variable

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    await page.click(flowContainerPage.contentfulCheckbox);
+    await Promise.all([
+      page.waitForResponse(response => response.url()
+        .includes('width_33') && response.status() === 200),
```
should be moved to a variable

TatsianaMiraniuk commented on playwright/pages/flow-container-page.ts:
```diff
@@ -0,0 +1,83 @@
+    this.selectImageButton = 'button[aria-label="Select Image"]';
+    this.editImageButton = 'button[aria-label="Edit Image"]';
+    this.squareImageButton = '#gridlistboxinner > div:nth-child(1) > div > button';
+    this.accordionInPicker = 'button[type="Submit"]';
```
@pomi ,

please rename - Accordion doesn't reflect Submit action

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    await page.click(flowContainerPage.addFlowContainerButton);
+    await page.click(flowContainerPage.accordionCheckbox);
+    await page.click(flowContainerPage.contentfulCheckbox);
+    await Promise.all([
```
@pomi ,

what that helps to do here? width_33 is FlowContainer ID - it looks not related to the steps we're doing: Component Selector is open, we uncheck Accordion, then check Contentful Card

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    await Promise.all([
+      page.waitForResponse(response => response.url()
+        .includes('width_33') && response.status() === 200),
+      await page.click(flowContainerPage.accordionInPicker),
```
@pomi ,

if there are more than one element found by flowContainerPage.accordionInPicker - the test is not failing? How Playwright handles that? It uses first found?

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    const img = defaultContainer.locator('img');
+    expect.soft(await img.getAttribute('alt')).toEqual('alt-test');
+    expect.soft(await img.getAttribute('src')).toContain(flowContainerPage.imageTwoName);
+    expect.soft(await page.locator(flowContainerPage.accordionPlusButton).isVisible()).toBeTruthy();
```
@pomi ,

please confirm that the text added in the Accordion is visible on preview (both Title and Body)

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+        .includes('flowContainer') && response.status() === 200),
+      page.click(flowContainerPage.submitButton),
+    ]);
+    const defaultContainer = page.locator(flowContainerPage.flowContainerDefault);
```
@pomi ,

that and below are repeated 3 times. Can it be moved to a global level?

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    await page.click('div[data-item-id="LandscapeLinkableImage"]');
+    const defaultContainer = page.locator(flowContainerPage.flowContainerDefault);
+    const img = defaultContainer.locator('img');
+    expect.soft(await img.getAttribute('alt')).toEqual('alt-test');
```
should be a variable instead of hardcoding

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    const img = defaultContainer.locator('img');
+    expect.soft(await img.getAttribute('alt')).toEqual('alt-test');
+    expect.soft(await img.getAttribute('src')).toContain(flowContainerPage.imageTwoName);
+    await page.locator('#flowContainer > div a >> img').click();
```
should be a variable

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    expect.soft(await img.getAttribute('src')).toContain(flowContainerPage.imageTwoName);
+    await page.locator('#flowContainer > div a >> img').click();
+    await page.click(flowContainerPage.editImageButton);
+    await flowContainerPage.typeText('#link-href', 'new_link', 'flowContainer', 'button[aria-label="Submit"]');
```
should be the variables instead of hardcoding

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    await page.locator('#flowContainer > div a >> img').click();
+    await page.click(flowContainerPage.editImageButton);
+    await flowContainerPage.typeText('#link-href', 'new_link', 'flowContainer', 'button[aria-label="Submit"]');
+    expect.soft(await page.locator('#flowContainer > div a').getAttribute('href')).toEqual('/' + 'new_link' + '/');
```
should be the variables instead of hardcoding

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+    const defaultContainer = page.locator(flowContainerPage.flowContainerDefault);
+    const img = defaultContainer.locator('img');
+    expect.soft(await img.getAttribute('alt')).toEqual('alt-test');
+    expect.soft(await img.getAttribute('src')).toContain(flowContainerPage.imageTwoName);
```
@pomi ,

imagePathRegex should be checked 
`const imagePathRegex = new RegExp("images\/pages" + pagePath + "[a-zA-Z0-9]+\/" + imageName, "");`
please see 
https://github.com/johnsonandjohnson/Bodiless-JS/pull/584/files#diff-abfd3f9753c21f575bdae486babc4d2f7fd5129967c231e337515b047ea149c2R29

https://github.com/johnsonandjohnson/Bodiless-JS/pull/584/files#diff-abfd3f9753c21f575bdae486babc4d2f7fd5129967c231e337515b047ea149c2R99

===
the following should be added on global level (commented about it in the Migration to Playwright PR)
https://github.com/johnsonandjohnson/Bodiless-JS/pull/584/files#diff-abfd3f9753c21f575bdae486babc4d2f7fd5129967c231e337515b047ea149c2R100

TatsianaMiraniuk commented on playwright/tests/smoke/flowContainer.spec.ts:
```diff
@@ -0,0 +1,148 @@
+
+  test('checking the added components in Preview Mode', async () => {
+    await flowContainerPage.togglePreviewMode();
+    expect.soft(await page.locator(flowContainerPage.swapComponentButton).isVisible()).toBeFalsy();
```
@pomi ,

before checking that Add, Copy, Swap, Delete are not available in Preview Mode, do a click on any added component - to prove that Local Context Menu with that buttons doesn't show up.
Originally that buttons were available in Editor Menu, and not in Local Context Menu. that's why the 'click' step is not in the original test cases

Commit:
Merge branch 'pw_poc' into AESQ-2883

Commit:
preliminary commit

lucas-varela committed:
fix: Fix Content Library hover indicator missing (#1786)

Co-authored-by: Heidi Vanyo <58534559+hvanyo@users.noreply.github.com>

hvanyo committed:
fix(vital-demo): fix logo & favicons (#1788)

* Add Hero component to cx-image

* feature(cx-structuredata): Provide Structure Data Package

* feature(cx-structuredata): Fix Lint Errors And Warnings

* Add Hero component to cx-image

* Move cxHero to its own package

* Rename cx-hero to vital-hero

* Update vital-hero docs

* Fix broken build

* feature(vital-structuredata): Rename CX Structure Data To Vital

* feature(vital-structuredata): Update Bodiless Deps On StructureData Package

* feature(vital-structuredata): Remove Hero From CX Image After Merge Conflicts

* feature(vital-structuredata): Update StructureData Package Version

* feature(vital-youtube): Add Vital YouTube Package

* feature(vital-youtube): Add YT Flow Container Variations

* feature(cx-flowcontainer): Import YT Variations On Flow Container

* feature(vital-youtube): Rename Vital YT Docs Json

* feature(vital-youtube): Update YT Default Settings

* feature(vital-youtube): Restructure Vital YouTube

* feature(vital-structuredata): Add SD Provider To Page Layout

* Update vital-hero package version

* Update vital packages to v1.0.0-beta.3

* Fix build breaking

* Add video hero variation

* Update cx metions to vital

* Update hero docs

* Rename __cxstarter__Hero to __vital__Hero

* feat(vital-card): Hero Card

* move to upgrade branch

* token update

* feat(vital-hero): Update Hero, SD, And YT Packages Versions

* refactoring

* fixing hmr bundle rebuild issue in __vital__ package

* vitalCard token

* docs and design fix

* Remove vital-hero package

* token refactoring

* use nodeKey constant

* adjust styles

* Remove Hero component from __vital__

* Move Hero examples to Images and Video pages

* Update vital-youtube tokens

* token update

* Update vital-youtube tokens

* Renamte vital-structuredata to bodiless-schema-org

* Move video styleguide page to vital-demo

* Card flow container

* Card flow container

* bring over vitalRichText.BasicNoLink from pr#1498, replacing Default on RichText

* Meta title

* remove hero card added by accident

* Meta term update and variants

* Meta term update and variants

* remove hero from flowcontainer since it`s adding to hero flowcontainer

* adding hero card back

* update variants

* add HeroCard to topcontent chameleon

* fix(vital-card): identical variations in component

* fix(vital-card): HERO  TITLE = H1 , DESCRIPTION = H4

* Cards with no cta are not fully clickable / Hero card image is clickable

* fix(vital-card): Hero card image needs eager

* Add schema info to vital-youtube docs

* Fix video schema showing error with no API key set

* Add fullscreen to Hero video token

* Update hero images

* Add node key to hero images

* Make hero images full-width after upload

* Update vital-image docs

* Fix lint error

* Update hero node keys

* Add missing comma in vitalImage

* Update packages to 1.0.0-beta.9

* fix(gatsby-theme-bodiless): Fix loading eager not applied to gatsby images

* Update youtube api key env var name

* fix(vital-card): token updates

* fix(vital-card): token updates

* feat(vital-image): Vital-Image is static

* address feedback

* Address additional feedbacks

* remove useless comments

* Remove gatsby-plugin-image from site __vital__

* remove duplicated nodekey and withoutHydraion due to varydesign

* fix image loading race condition

* Fixed gatsby images served as webp

* Fix Images disappearing

* fix - the local context menu doesn't appear

* Add gatsby plugin to allow unprefixed env vars

* Update packages to 1.0.0-beta.11

* fix(vital-card): Whole CTA buttons is clickable - additional layout fix

* Update image flow continaer tokens

* Update packages to v1.0.0-beta.12

* fix(vital-card): vital card static

* Add static code to vital-youtube

* Fix static card code not available on production

* Update dependencies to 1.0.0-beta.13

* Update dependency versions

* Update package dependencies

* Update package versions

* Update packages to v1.0.0-beta.18

* fix bodiless/hydration tests

* fix logo

* Change Logo & Favicon

* remove logos from __vital__ that aren't used

* change name without spaces

Co-authored-by: Lucas Varela <lvarelab@its.jnj.com>
Co-authored-by: Guilherme Almeida Zeni <gzeni@ciandt.com>
Co-authored-by: Guilherme Zeni <gzeni@its.jnj.com>
Co-authored-by: Dewen Li <dli94@its.jnj.com>
Co-authored-by: Marco Pagliarulo <mpaglia1@its.jnj.com>
Co-authored-by: Alexey Semirski <al.semirski@gmail.com>

lucas-varela committed:
fix(release): Make schema-org and vital-youtube packages public (#1795)

Commit:
Merge branch 'pw_poc' into AESQ-2883

Commit:
Tests are updated according to PR comments

Commit:
Tests are updated according to PR comments

Commit:
Tests are updated according to PR comments

Commit:
chore(release): publish v1.0.0-rc.1

Commit:
Release 1.0.0-rc.1 package.json and lock file updates.

dewen committed:
Merge pull request #1797 from johnsonandjohnson/release

chore: Release v1.0.0-rc.1

lucas-varela committed:
ci: Release RC versions on prerelease (#1798)

wodenx committed:
docs: Provide API docs pattern, docs for fclasses and vital-flowcontainer (#1683)

## Changes
Implement typedoc patterns for fclasses and vital-fowcontainer
document best practices.

Commit:
fix(docs): fixing failed gh-page publishing

dewen committed:
Merge pull request #1800 from dewen/fix/docs-build

fix(docs): fixing failed gh-page publishing

Asemirski committed:
ci: canary release improvements (#1784)

Commit:
feat(playwright): Migration from cypress to playwright (#1487)

* Migration from cypress to playwright

* Migration from cypress to playwright, updated

* Migration from cypress to playwright, updated

* typeText method was removed from pdp.spec.ts

* fix: add container selectors on flow-container page

* Use area-label insead of className

* FlowContainer component: allow id prop

* fixed all PR comments

* AESQ-8348, package-lock.json updated

* AESQ-8348, obsolete files were deleted

* AESQ-2883, viewport removed from all files

* AESQ-8348, obsolete files were deleted

* Preliminary commit

* AESQ-8348, fully reworked tests, 1-1 to cypress

* removed obsolete files

* Files updates to comply with MR comments

* Fixed according to MR

* Fixed according to MR

* Fixed according to MR

* Fixed viewport for chromium

* fixed tests according to PR comments

* preliminary commit

* Updates according to PR comments

* fix: add container selectors on flow-container page

* Use area-label insead of className

* FlowContainer component: allow id prop

* package-lock.json updated with playwright 1.22 version

* package-lock.json updated with playwright 1.22 version

* changed number of commits to 1 in commit history check

* limit number of workers for playwright

* cards.spec.ts was excluded from test suite

* fixed commit history comparison

Co-authored-by: pomi <AQ1sw@DE3fr$GT5>
Co-authored-by: tmiraniuk <tmiraniu@its.jnj.com>
Co-authored-by: Konstantin Mishur <kmishur@its.jnj.com>
Co-authored-by: Heidi Vanyo <58534559+hvanyo@users.noreply.github.com>

lucas-varela committed:
fix(editors): Fix hover menu closing on click (#1799)

* fix(editors): Fix hover menu closing on click

* Fix toggle trying to select an invalid range

Co-authored-by: Heidi Vanyo <58534559+hvanyo@users.noreply.github.com>

wodenx committed:
fix(fclasses): Improve mobx warning. (#1794)

misterjones committed:
docs(bodiless-documentation): Update release process (#1482)

* Add "Normal Release Process" section.
  * This includes Pre-Releases.
* Add "Fully Manual Release Process" section.

Co-authored-by: Chris Oden <coden@its.jnj.com>
Co-authored-by: Dewen Li <dewen.sinocrest@gmail.com>

misterjones committed:
docs(bodiless-schema-org, vital-card, vital-youtube): Vital Hero (#1792)

* Fix generation/build of documentation.
  * Add `bodiless-schema-org`, `vital-card`, and `vital-youtube`
    packages to Test Site and Vital Demo Site `package.json` files.
  * Add `getDocs.js` files to `bodiless-schema-org`, `vital-card`, and
    `vital-youtube` packages.
  * Update `package.json` files of `bodiless-schema-org`, `vital-card`,
    and `vital-youtube` to include `getDocs.js` and `@bodiless/cli`.
* Update doc structure
* Update `bodiless.docs.json` files.
  * Rename main doc file to `README.md`.
* Update package `README`s.
* Add `assets` folders with images.
* Update docs with Vital Hero info.
  * Update Vital Card, Image, and YouTube documentation with Hero
    details.
  * Clean up Bodiless YouTube Component page.
  * Update Schema page.
* Fix broken links in Vital Demo style guide.
* Break Product and Video Schema sections into pages
  * Break the "Product Schema" and "Video Schema" sections out of the
    main "Schema Component" page, and into their own pages under the
    Schema section.
* Fill out Vital Card page.
  * Add info block about Card height.

Co-authored-by: Heidi Vanyo <58534559+hvanyo@users.noreply.github.com>
Co-authored-by: Dewen Li <dewen.sinocrest@gmail.com>

hvanyo committed:
feat(vital-navigation): Add Breadcrumbs (#1778)

* feat(vital-link): add buttons

* add breadcrumbs

* add breadcrumbs

* update buttons styles

* fix breadcrubm

* Update Buttons with hover

* update button

* fix spacing

* move buttons to vital-buttons

* add withBreadcrumbstore

* Move Link -> Buttons in styleguide

* fix nodekeys

* Working buttons

* move WTB separate file

* Rename to use ButtonClean/asButtonToken

* fix lint error

* fix shadowing

* fix vitalBreadcrumbs.Default

* fix link size

* remove button

* remove more buttons changes

* remove lint error

* add TextPrimaryInteractiveHover

* fix(vital-navigation): missing node collection for breadcrumb

* fix(vital-navigation): For a non-Parent menu with submenu items -- breadcrumb link is wrong

* fix(vital-navigation): For a non-Parent menu with submenu items -- breadcrumb link is wrong

* fix(vital-navigation): For a non-Parent menu with submenu items -- breadcrumb link is wrong

* fix(vital-navigation): For a non-Parent menu with submenu items -- breadcrumb link is wrong

* fix(vital-navigation): For a non-Parent menu with submenu items -- breadcrumb link is wrong; FinalTrail styling fix

* fix(vital-navigation): last item bold

* fix(vital-navigation): last item bold

* fix(vital-navigation): last item bold

Co-authored-by: Dewen Li <dli94@its.jnj.com>

hvanyo committed:
feat(vital-buttons): Add Buttons (#1781)

* feat(vital-buttons): add buttons


Co-authored-by: Dewen Li <dewen.sinocrest@gmail.com>

Asemirski committed:
fix: checkout to merge commit on pull_request_target event (#1808)

Commit:
Merge branch 'main' into AESQ-2883

lucas-varela committed:
fix(lists): Fix list context menu not opening on tab (#1771)

* Fix list context menu not opening on tab

* Activate list item's context on creation

Co-authored-by: Heidi Vanyo <58534559+hvanyo@users.noreply.github.com>

lucas-varela committed:
feat(search): Add @bodiless/vital-search (#1628)

* WIP: add vital-search package

* Fix vital-search dependency versions

* Update vital-search git url

* Update vital-search docs config file

* Add bodiless search reexports into vital search

* Remove vital-layout dependency from vital-search

* Update search box styles

* Add search page to vital-demo

* Update search configuration lookup

* Update vital-search docs

* Update vital-search styles

* Update packages to 1.0.0-beta.12

* Add minor style fixes

* Fix typo in bodiless-search docs

* Add mobile styles to the search component

* Update vital search dependencies

* Move search components into vital-search

* Add copyright message to vital-serch files

* Update vital-search version

* Refactor vital-search component structure

* Update vital-search imports on vital-layout

* Rename SearchResult to SearchResults

* Rename SearchResultItem to SearchResult

* Update SearchLayoutComponents type

* Move search context to a separate layout token

* Update vital-search tokens

* Update Search page template structure

* Remove ResponsiveBreakpoints helper

* Update vital-search to 1.0.0-beta.18

* Update vital-search token exports

* Fix __vital__ site build breaking

* Remove styleguide pages from vital-demo search

* Add H1Link through H5Link typography tokens

* Fix homepage search result not having a title

* Add colored backgrounds to search suggestion rows

* Fix SearchTool test

* Fix broken home page href in search results

* Fix vital-search colors missing on static sites

* Update vital-search to v1.0.0-rc.1

* Add missing search secondary color token

Co-authored-by: Heidi Vanyo <58534559+hvanyo@users.noreply.github.com>

hvanyo committed:
fix(vital-cards): update cards with buttons & misc fixes (#1805)

* Fix set of issues with cards/buttons & other vital design issues.

Commit:
Merge branch 'main' into AESQ-2883
